Resolve imports with query params

dons20 opened the pull request:
I've been using vite for some time and I needed to have my web worker properly resolved with eslint. Workers in vite require me to add a `?worker` to the end of the import path which breaks this resolver. I made this little change locally and it works great for this use case, so maybe this is something you'd like to include?

Let me know what you think.

dons20 committed:
Resolve imports with query params

laysent commented:
I am wondering if you could make this behavior optional? i.e. user can use config to enable/disable it.
I think `?` might be used in other scenarios as well (perhaps in combination with some webpack loader), not very sure if this behavior is what users always want.

dons20 committed:
Merge branch 'laysent:master' into patch-1

dons20 committed:
Made strip query feature optional

dons20 commented:
Totally forgot I had this done. I've pushed a change to make this feature optional as it rightly should be 😅 . If this looks good then all it will need is updated documentation and tests

dons20 committed:
Updated tests for new parameter
Updated usage of deprecated `substr` to `substring`
Updated some tests to use `path.normalize` for windows support
